markdown-toc.el

```diff
@@ -175,6 +175,14 @@ Return the end position if it exists, nil otherwise."
       markdown-toc--to-markdown-toc
       markdown-toc--compute-full-toc))
 
+(defun markdown-toc--delete-toc ()
+  "Delets a TOC."
+  (let ((region-start (markdown-toc--toc-start))
+         (region-end   (markdown-toc--toc-end)))
+    (delete-region region-start (1+ region-end))
+    (when replace-toc-p
```

Comment:
This should be removed.
`replace-toc-p` does not exist in this refactored function.

I think I did not know at the time the save-excursion routines...
Maybe wrap this with `save-excursion` to not disturb the user.